Added an item with github source authentication to the dotnet CLI in the instructions for GitHub Packages

Falcion opened the pull request:
<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to an issue) and what changes you've made, we can triage your pull request to the best possible team for review.
-->

### Why:

The article itself, although contains information on how to authenticate `github` source for NuGet integration with GitHub Packages still indicate GitHub Actions in the title, which can confuse a user who is working with GitHub Packages for the first time and is trying to upload their project manually (again using `dotnet` CLI)

Example of this experience is this discussion: https://github.com/orgs/community/discussions/49186

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Update guide in [Publishing a package](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#publishing-a-package) and [Publishing a package using a nuget.config file](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#publishing-a-package-using-a-nugetconfig-file) and add step about authenticating GitHub Packages for user:

```markdown
1. Authentificate GitHub Packages with the `dotnet` command-line interface (CLI). Replace `OWNER` with your username or company name.

    ```shell
    dotnet nuget add source --username OWNER --password ${ secrets.GITHUB_TOKEN } --store-password-in-clear-text --name github "https://nuget.pkg.github.com/OWNER/index.json"
    ```
```
### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

Falcion committed:
docs(working-with-a-github-packages-registry): add instructions to registry github source

Falcion committed:
Merge branch 'main' into patch-1

Falcion committed:
Merge branch 'main' into patch-1

thyeggman commented on content/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry.md:
````diff
@@ -129,6 +129,12 @@ If you don't already have a {% data variables.product.pat_generic %} to use for
    dotnet pack --configuration Release
    ```
 
+1. Authentificate GitHub Packages with the `dotnet` command-line interface (CLI). Replace `OWNER` with your username or company name.
````
```suggestion
1. Authenticate to GitHub Packages with the `dotnet` command-line interface (CLI). Replace `OWNER` with your username or company name, and `YOUR_GITHUB_PAT` with your personal access token.
```

thyeggman reviewed (commented):
This step already exists in the "Authenticating to GitHub Packages" section of the documentation, although it is formatted for use in a GitHub Actions workflow.  I think the confusion is understandable, though.

I would suggest removing these two changes and moving the new part up to the above section. Use `YOUR_GITHUB_PAT` to match other examples instead of `${{ secrets.GITHUB_TOKEN }}`.  Mention that it is a command that can be run directly from the command line using the `dotnet` CLI instead of manually configuring a `nuget.config` file.  Also check the spelling of "Authenticate" 👍

Falcion committed:
Updated PR with @thyeggman suggestion

Co-authored-by: Jacob Wallraff <thyeggman@github.com>

Falcion committed:
chore(fix): tweaks and fixes

1. Removed duplicates about auth;
2. Moved authentification with CLI into the "Publishing a package"
3. Replaced `${{ secrets.GITHUB_TOKEN }}` with `YOUR_GITHUB_PAT`

Falcion committed:
chore(lintfix): fix lint according to lint checker

Falcion committed:
chore(lintfix): fix lint according to lint checker (md009)

Falcion committed:
chore(lintfix): fix lint according to lint checker (md009)

Falcion commented:
I apologize for the "spammy" commits with fixes and adjustments to your answer.

As was mentioned in the discussion, I removed the “duplicate” authorizations, changed `${{ secrets.GITHUB_TOKEN }}` to `YOUR_GITHUB_PAT`, combined and moved the authorization instructions to the “Publish a package” block, if something else required, I'll be happy to try to fix/finish it.

nguyenalex836 commented:
@Falcion Thank you so much for that last batch of adjustments! 💛 I'll update the branch and get this merged once tests are passing 🍏

nguyenalex836 committed:
Merge branch 'main' into patch-1
